### DISCUSSION TITLE
Can't succeed future tasks when non existant in previous dag runs

### DISCUSSION THREAD

**Discussion by alejandrofm** [apache/airflow#35731] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Airflow 2.7.2
When the task was non-existent for the DAG "cleared" period, 
I cannot succeed future tasks, only current
![image](https://github.com/apache/airflow/assets/19591174/1bf254ac-1f8c-4081-822d-f5879317fd96)

### What you think should happen instead

I should be able to succeed in any kind of task in the future or past.

### How to reproduce

On a dag run currently have 5 tasks running from November 1.
I cleared October to re-run it (where there were only 4 tasks)
These tasks appeared now and want to set them to success from the first to future.

### Operating System

Airflow 2.7.2 Python3.10 official image

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Same as https://github.com/apache/airflow/discussions/35730?converting=1 -> this is DAG versioning feature that we do not have. If you change DAG structure significantly, you should make a copy of it and change DAG id - effectively having 2 different DAGs for different periods.
